src/metrics/chaoss.ts

```diff
@@ -411,4 +411,54 @@ FROM
         lines,
       }
     });
-  };
\ No newline at end of file
+  };
+
+  export const chaossNewContributors = async (config: QueryConfig) => {
+    config = getMergedConfig(config);
+    const whereClauses: string[] = ["type = 'PushEvent'"];
+    const repoWhereClause = getRepoWhereClauseForClickhouse(config);
+    if (repoWhereClause) whereClauses.push(repoWhereClause);
+    // whereClauses.push(getTimeRangeWhereClauseForClickhouse(config));
+    const endDate = new Date(`${config.endYear}-${config.endMonth}-1`);
+    endDate.setMonth(config.endMonth);  // find next month
+    
+    const sql = `
+    SELECT
+      id,
+      argMax(name, time) AS name,
+      SUM(count) AS total_count,
+      ${getGroupArrayInsertAtClauseForClickhouse(config, { key: 'new_contributors', value: 'count' })}
+    FROM
+    (
+      SELECT
+        ${getGroupTimeAndIdClauseForClickhouse(config, 'repo')},
+        count() as count
+        FROM(
+         SELECT min(created_at)as created_at,org_id,repo_id,name
+         FROM(
+          SELECT created_at,org_id,repo_id,arrayJoin(push_commits.name) as name
+          FROM github_log.events
+          WHERE ${whereClauses.join(' AND ')}
+           )AS t
+         GROUP BY org_id,repo_id,name
+      )
+      GROUP BY id, time
+      HAVING time >= toDate('${config.startYear}-${config.startMonth}-1') AND time <= toDate('${endDate.getFullYear()}-${endDate.getMonth() + 1}-1')
+      ${config.limit > 0 ? `ORDER BY count DESC LIMIT ${config.limit} BY time` : ''}
+    )
+    GROUP BY id
+    ORDER BY new_contributors[-1] ${config.order}
+    FORMAT JSONCompact`;
+    
+    const result: any = await clickhouse.query(sql);
+    return result.map(row => {
+    const [ id, name, total_count, count ] = row;
+    return {
+      id,
+      name,
+      total_count,
+      count,
+      ratio: count.map(v => `${(v*100/total_count).toPrecision(2)}%`),
```

Comment:
In the CHAOSS metrics description: https://chaoss.community/metric-new-contributors/ , we don't really need a `ratio` filed in the result. Instead, we should return a `detail` field with all the new contributors name and how many commits did they make in the time period.